pkg/flare/writer.go

```diff
@@ -0,0 +1,130 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+package flare
+
+import (
+	"bufio"
+	"errors"
+	"io/ioutil"
+	"os"
+
+	log "github.com/cihub/seelog"
+)
+
+//RedactingWriter is a writer that will redact content before writing to target
+type RedactingWriter struct {
+	target    *os.File
+	targetBuf *bufio.Writer
+	perm      os.FileMode
+	r         []replacer
+}
+
+//NewRedactingWriter instantiates a RedactingWriter to target with given permissions
+func NewRedactingWriter(t string, p os.FileMode, buffered bool) (*RedactingWriter, error) {
+	err := ensureParentDirsExist(t)
+	if err != nil {
+		return nil, err
+	}
+
+	f, err := os.OpenFile(t, os.O_RDWR|os.O_CREATE, p)
+	if err != nil {
+		return nil, err
+	}
+
+	var b *bufio.Writer
+	if buffered {
+		b = bufio.NewWriter(f)
+	}
+
+	return &RedactingWriter{
+		target:    f,
+		targetBuf: b,
+		perm:      p,
+		r:         []replacer{},
+	}, nil
+}
+
+//RegisterReplacer register additional replacers to run on stream
+func (f *RedactingWriter) RegisterReplacer(r replacer) {
+	f.r = append(f.r, r)
+}
+
+//WriteFromFile will read contents from file and write them redacted to target
+func (f *RedactingWriter) WriteFromFile(filePath string) (int, error) {
+
+	if _, err := os.Stat(filePath); err != nil {
+		if os.IsNotExist(err) {
+			log.Warnf("the specified path: %s does not exist", filePath)
+		}
+
+		return 0, err
+	}
+
+	data, err := ioutil.ReadFile(filePath)
```

Comment:
Should we truncate the file in case we're rewriting the same file and the redacted content is shorter (which might be the case with secrets that we replace with `*****` for example) ?

Reply:
Good point, done!